### PR TITLE
[Bug](MaterializedView) Fix bug of light schema change do not set right unique id cause MV coredump

### LINKED ISSUE #11334: [Bug] core dump on compaction
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master

### What's Wrong?
```sql
CREATE TABLE test_sys_materialized_view_2_test_delete_with_mv_db.test_sys_materialized_view_2_test_delete_with_mv_tb ( k0 BOOLEAN NOT NULL, k1 TINYINT NOT NULL, k2 SMALLINT NOT NULL, k3 INT NOT NULL, k4 BIGINT NOT NULL, k5 LARGEINT NOT NULL, k6 DECIMAL(9,3) NOT NULL, k7 CHAR(5) NOT NULL, k8 DATE NOT NULL, k9 DATETIME NOT NULL, k10 VARCHAR(20) NOT NULL, k11 DOUBLE NOT NULL, k12 FLOAT NOT NULL )  PARTITION BY RANGE(k1) ( PARTITION p1 VALUES LESS THAN ("-10"), PARTITION p2 VALUES LESS THAN ("0"), PARTITION p3 VALUES LESS THAN ("10"), PARTITION p4 VALUES LESS THAN ("20") ) DISTRIBUTED BY HASH(k4, k5) BUCKETS 10 properties("replication_num" = "1");

CREATE MATERIALIZED VIEW test_sys_materialized_view_2_test_delete_with_mv_index AS SELECT k0, max(k9), min(k11), sum(k12), count(k10), hll_union(hll_hash(k7)), bitmap_union(to_bitmap(k2)) from test_sys_materialized_view_2_test_delete_with_mv_db.test_sys_materialized_view_2_test_delete_with_mv_tb group by k0;


load baseall.txt

DELETE FROM test_sys_materialized_view_2_test_delete_with_mv_db.test_sys_materialized_view_2_test_delete_with_mv_tb PARTITION p3 WHERE k0 = "0";

```

```cpp

F0729 14:34:34.184578 36127 column.h:189] Method insert_many_fix_len_data is not supported for HLL

*** Query id: 0-0 ***
*** Aborted at 1659076474 (unix time) try "date -d @1659076474" if you are using GNU date ***
*** Current BE git commitID: fd0e2d3 ***
*** SIGABRT unkown detail explain (@0x1f500008bea) received by PID 35818 (TID 0x7f8e62d4f700) from PID 35818; stack trace: ***
 0# doris::signal::(anonymous namespace)::FailureSignalHandler(int, siginfo_t*, void*) at /root/doris-release/be/src/common/signal_handler.h:420
 1# 0x00007F8ED2661920 in /lib64/libc.so.6
 2# __GI_raise in /lib64/libc.so.6
 3# abort in /lib64/libc.so.6
 4# 0x0000559A41D0F289 in /home/disk1/palo-qa/teamcity/local/9137/PALO-BE/be/lib/doris_be
 5# 0x0000559A41D0489D at src/logging.cc:1650
 6# google::LogMessage::SendToLog() at src/logging.cc:1607
 7# google::LogMessage::Flush() at src/logging.cc:1477
 8# google::LogMessageFatal::~LogMessageFatal() at src/logging.cc:2227
 9# doris::vectorized::IColumn::insert_many_fix_len_data(char const*, unsigned long) at /root/doris-release/be/src/vec/columns/column.h:189
10# doris::segment_v2::BitShufflePageDecoder<(doris::FieldType)15>::read_by_rowids(unsigned int const*, unsigned long, unsigned long*, COW<doris::vectorized::IColumn>::mutable_ptr<doris::vectorized::IColumn>&) at /root/doris-release/be/src/olap/rowset/segment_v2/bitshuffle_page.h:423
11# doris::segment_v2::FileColumnIterator::read_by_rowids(unsigned int const*, unsigned long, COW<doris::vectorized::IColumn>::mutable_ptr<doris::vectorized::IColumn>&) at /root/doris-release/be/src/olap/rowset/segment_v2/column_reader.cpp:781
12# doris::segment_v2::SegmentIterator::_read_columns_by_rowids(std::vector<unsigned int, std::allocator<unsigned int> >&, std::vector<unsigned int, std::allocator<unsigned int> >&, unsigned short*, unsigned long, std::vector<COW<doris::vectorized::IColumn>::mutable_ptr<doris::vectorized::IColumn>, std::allocator<COW<doris::vectorized::IColumn>::mutable_ptr<doris::vectorized::IColumn> > >*) at /root/doris-release/be/src/olap/rowset/segment_v2/segment_iterator.cpp:1015
13# doris::segment_v2::SegmentIterator::next_batch(doris::vectorized::Block*) at /root/doris-release/be/src/olap/rowset/segment_v2/segment_iterator.cpp:1121
14# doris::BetaRowsetReader::next_block(doris::vectorized::Block*) at /root/doris-release/be/src/olap/rowset/beta_rowset_reader.cpp:205
15# doris::vectorized::VCollectIterator::Level0Iterator::_refresh_current_row() at /root/doris-release/be/src/vec/olap/vcollect_iterator.cpp:209
16# doris::vectorized::VCollectIterator::Level0Iterator::init() at /root/doris-release/be/src/vec/olap/vcollect_iterator.cpp:194
17# doris::vectorized::VCollectIterator::build_heap(std::vector<std::shared_ptr<doris::RowsetReader>, std::allocator<std::shared_ptr<doris::RowsetReader> > >&) at /root/doris-release/be/src/vec/olap/vcollect_iterator.cpp:72
18# doris::vectorized::BlockReader::_init_collect_iter(doris::TabletReader::ReaderParams const&, std::vector<std::shared_ptr<doris::RowsetReader>, std::allocator<std::shared_ptr<doris::RowsetReader> > >*) at /root/doris-release/be/src/vec/olap/block_reader.cpp:63
19# doris::vectorized::BlockReader::init(doris::TabletReader::ReaderParams const&) at /root/doris-release/be/src/vec/olap/block_reader.cpp:135
20# doris::Merger::vmerge_rowsets(std::shared_ptr<doris::Tablet>, doris::ReaderType, doris::TabletSchema const*, std::vector<std::shared_ptr<doris::RowsetReader>, std::allocator<std::shared_ptr<doris::RowsetReader> > > const&, doris::RowsetWriter*, doris::Merger::Statistics*) at /root/doris-release/be/src/olap/merger.cpp:114
21# doris::Compaction::do_compaction_impl(long) at /root/doris-release/be/src/olap/compaction.cpp:170
22# doris::Compaction::do_compaction(long) at /root/doris-release/be/src/olap/compaction.cpp:122
23# doris::BaseCompaction::execute_compact_impl() at /root/doris-release/be/src/olap/base_compaction.cpp:70
24# doris::Compaction::execute_compact() at /root/doris-release/be/src/olap/compaction.cpp:60
25# doris::Tablet::execute_compaction(doris::CompactionType) at /root/doris-release/be/src/olap/tablet.cpp:1571
26# std::_Function_handler<void (), doris::StorageEngine::_submit_compaction_task(std::shared_ptr<doris::Tablet>, doris::CompactionType)::{lambda()#1}>::_M_invoke(std::_Any_data const&) at /var/local/ldb-toolchain/include/c++/11/bits/std_function.h:291
27# doris::ThreadPool::dispatch_thread() at /root/doris-release/be/src/util/threadpool.cpp:548
28# doris::Thread::supervise_thread(void*) at /root/doris-release/be/src/util/thread.cpp:409
29# start_thread in /lib64/libpthread.so.0
30# clone in /lib64/libc.so.6
```

### What You Expected?

fix

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by HappenLee**
# Proposed changes

Issue Number: close #11334 

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by yiguolei — DISMISSED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/analysis/MVColumnItem.java [hunk 1]
```diff
@@ -108,6 +108,7 @@ public Column toMVColumn(OlapTable olapTable) throws DdlException {
             Preconditions.checkNotNull(defineExpr != null);
             Column result = new Column(name, type, isKey, aggregationType, ColumnDef.DefaultValue.ZERO, "");
             result.setDefineExpr(defineExpr);
+            result.setUniqueId(olapTable.incAndGetMaxColUniqueId());
```
Comment (SWJTU-ZhangLei) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/MVColumnItem.java:111]: In the future, this change will cause problem when we want to support mv do light schema change and olapTable.incAndGetMaxColUniqueId() need to write editlog for duration.   
the commit  can justly use temporaryly.

Comment (SWJTU-ZhangLei) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/MVColumnItem.java:111]: could you try like this in   MaterializedViewHandler.java  checkAndPrepareMaterializedView()

        for (Column column : rollupSchema) {
            column.setUniqueId(Column.COLUMN_UNIQUE_ID_INIT_VALUE);
        }

Comment (BiteTheDDDDt) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/MVColumnItem.java:111]: > could you try like this in MaterializedViewHandler.java checkAndPrepareMaterializedView()
> 
> ```
>     for (Column column : rollupSchema) {
>         column.setUniqueId(Column.COLUMN_UNIQUE_ID_INIT_VALUE);
>     }
> ```

Hi, Do you can review #10792? I set uniqid also.

Comment (SWJTU-ZhangLei) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/MVColumnItem.java]: @ the change will override yours

> > could you try like this in MaterializedViewHandler.java checkAndPrepareMaterializedView()
> > ```
> >     for (Column column : rollupSchema) {
> >         column.setUniqueId(Column.COLUMN_UNIQUE_ID_INIT_VALUE);
> >     }
> > ```
> 
> Hi, Do you can review #10792? I set uniqid also.

 the change will override yours

**Review by swjtu-zhanglei — DISMISSED** [review]

File: regression-test/conf/regression-conf.groovy [hunk 2]
```diff
@@ -20,11 +20,11 @@
 // **Note**: default db will be create if not exist
 defaultDb = "regression_test"
 
-jdbcUrl = "jdbc:mysql://127.0.0.1:9030/?"
+jdbcUrl = "jdbc:mysql://127.0.0.1:9031/?"
```
Comment (BiteTheDDDDt) [inline regression-test/conf/regression-conf.groovy:23]: need reset

**Review by BiteTheDDDDt — APPROVED** [review]

**Pull request merged**